tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/effects/debugger_effects.ts

```diff
@@ -653,6 +657,88 @@ export class DebuggerEffects {
     );
   }
 
+  /**
+   * Listens to graph-op focus events.
+   *
+   * Load graph op info from the /graphs/op_info route when necessary.
+   *
+   * @returns the IDs of the stack frames of the op's creation for downstream
+   * consumption.
+   */
+  private loadGraphOpInfo(): Observable<string[]> {
+    return this.actions$.pipe(
+      ofType(graphOpFocused),
+      withLatestFrom(
+        this.store.select(getActiveRunId),
+        this.store.select(getLoadingGraphOps)
+      ),
+      filter(([actionData, runId, loadingOps]) => {
+        const {graph_id, op_name} = actionData;
+        return (
+          runId !== null &&
+          (loadingOps[graph_id] === undefined ||
+            loadingOps[graph_id][op_name] === undefined ||
+            loadingOps[graph_id][op_name] === DataLoadState.FAILED)
+        );
+      }),
+      tap(([actionData]) =>
+        this.store.dispatch(graphOpInfoRequested(actionData))
+      ),
+      mergeMap(([actionData, runId]) => {
+        const {graph_id, op_name} = actionData;
+        return this.dataSource.fetchGraphOpInfo(runId!, graph_id, op_name).pipe(
+          map((graphOpInfoResponse) => {
+            this.store.dispatch(graphOpInfoLoaded({graphOpInfoResponse}));
+            return graphOpInfoResponse.stack_frame_ids;
```

Comment:
nit: iirc, the pattern in most places is to separate `store.dispatch` calls into their own `tap()` operator.  This would be consistent with L684

Reply:
Done.